include/aws/auth/signing.h

```diff
@@ -53,10 +58,29 @@ AWS_EXTERN_C_BEGIN
  *      X-Amz-Algorithm,
  *      X-Amz-SignedHeaders
  *
- * In all cases, the signing result will tell exactly what header and/or query params to add to the request
+ * The signing result will tell exactly what header and/or query params to add to the request
  * to become a fully-signed AWS http request.
  *
- * These restrictions can be relaxed, if necessary, in the future.
+ *
+ * When using this signing function to sign chunks:
+ *
+ *   (1) The signable should be a wrapper around the chunk's input stream
+ *   (2) The most-recently-computed signature value (original request or most recent chunk) must be the
+ *       "previous-signature" property on the signable.
+ *
+ * The signing result will include the chunk's signature as the "signature" property.
+ *
+ *
+ * When using this function to sign events:
+ *
+ *   (1) The signable should be a wrapper around the event's payloads and headers.  TBD: forbid :date and calculate
+ *       it based on the config's date value?
```

Comment:
not 100% sure what this TBD is about, but I don't think you don't need to forbid "date".
If we always add "x-amz-date", then the vanilla "date" header is nothing special

Reply:
events can have headers and the signing process can add them.  They're not request headers nor h2 headers, they're weird things encoded into the event payload.